storage-transfer/src/main/java/com/google/cloud/storage/storagetransfer/samples/TransferFromAws.java

```diff
@@ -0,0 +1,129 @@
+
+    // A short description of this job
+    // String jobDescription = "Sample transfer job from S3 to GCS.";
+
+    // The name of the source AWS bucket to transfer data from
+    // String awsSourceBucket = "yourAwsSourceBucket";
+
+    // The name of the GCS bucket to transfer data to
+    // String gcsSinkBucket = "your-gcs-bucket";
+
+    // What day and time in UTC to start the transfer, expressed as an epoch date timestamp.
+    // If this is in the past relative to when the job is created, it will run the next day.
+    // long startDateTime =
+    //     new SimpleDateFormat("yyyy-MM-dd HH:mm:ss").parse("2000-01-01 00:00:00").getTime();
+
+    // The ID used to access your AWS account. Should be accessed via environment variable.
+    // String awsAccessKeyId = System.getenv("AWS_ACCESS_KEY_ID");
+
+    // The Secret Key used to access your AWS account. Should be accessed via environment variable.
+    // String awsSecretAccessKey = System.getenv("AWS_SECRET_ACCESS_KEY");
+
+    // Set up source and sink
+    TransferSpec transferSpec =
+        TransferSpec.newBuilder()
+            .setAwsS3DataSource(
+                AwsS3Data.newBuilder()
+                    .setBucketName(awsSourceBucket)
+                    .setAwsAccessKey(
+                        AwsAccessKey.newBuilder()
+                            .setAccessKeyId(awsAccessKeyId)
+                            .setSecretAccessKey(awsSecretAccessKey)))
+            .setGcsDataSink(GcsData.newBuilder().setBucketName(gcsSinkBucket))
+            .build();
+
+    // Parse epoch timestamp into the model classes
+    Calendar startCalendar = Calendar.getInstance();
+    startCalendar.setTimeInMillis(startDateTime);
+    // Note that this is a Date from the model class package, not a java.util.Date
+    Date startDate =
+        Date.newBuilder()
+            .setYear(startCalendar.get(Calendar.YEAR))
+            .setMonth(startCalendar.get(Calendar.MONTH) + 1)
+            .setDay(startCalendar.get(Calendar.DAY_OF_MONTH))
+            .build();
+    TimeOfDay startTime =
+        TimeOfDay.newBuilder()
+            .setHours(startCalendar.get(Calendar.HOUR_OF_DAY))
+            .setMinutes(startCalendar.get(Calendar.MINUTE))
+            .setSeconds(startCalendar.get(Calendar.SECOND))
+            .build();
+    Schedule schedule =
+        Schedule.newBuilder()
+            .setScheduleStartDate(startDate)
+            .setScheduleEndDate(startDate)
+            .setStartTimeOfDay(startTime)
+            .build();
+
+    // Set up the transfer job
+    TransferJob transferJob =
+        TransferJob.newBuilder()
+            .setDescription(jobDescription)
+            .setProjectId(projectId)
+            .setTransferSpec(transferSpec)
+            .setSchedule(schedule)
+            .setStatus(Status.ENABLED)
+            .build();
+
+    // Create a Transfer Service client
+    StorageTransferServiceClient storageTransfer = StorageTransferServiceClient.create();
```

Comment:
Do you want to move client creation before model construction?

Reply:
I can go either way on it. My idea was just having client creation right before the client is actually used, but I can move it up if you prefer

Reply:
I was thinking consistency, but I think your right in this case. You're only creating a client when you actually need it.